src/main.rs

```diff
@@ -109,6 +136,39 @@ const APP: () = {
         }
     }
 
+    #[task(binds = TIM3, resources = [sht_timer, sht, debug])]
+    fn sht_timer(ctx: sht_timer::Context) {
+        static mut STATE: ShtState = ShtState::StartMeasurement;
+
+        // Clear the interrupt flag
+        ctx.resources.sht_timer.clear_irq();
+
+        *STATE = match STATE {
+            ShtState::StartMeasurement => {
+                ctx.resources
+                    .sht
+                    .start_measurement(PowerMode::NormalMode)
+                    .expect("Failed to start measurement");
```

Comment:
Maybe add a prefix here for easier debugging if something goes wrong?

```suggestion
                    .expect("SHTCx: Failed to start measurement");
```

Reply:
Fixed.